### ISSUE TITLE
Provide a way to convert templates to a readable string representation

### ISSUE THREAD

**Issue by L7R7** [tfausak/burrito#12]
It would be great to have a possibility to convert a `Template` to a `String` that is the same as the input string. For example:
```hs
>>> let Just template = parse "http://user.example/~{name}"
>>> humanReadable template
"http://user.example/~{name}"
```
with a type signature like `humanReadable :: Template -> String`.

Maybe it makes sense to use `show` for that. Right now the `Show` instance for `Template` and the underlying datatypes are derived, so it's not really what I am looking for.

What do you think? Is that something you see as part of the library? If so, would you change the `Show` instance for `Template` to reflect that or would you prefer a separate function?

I would love to hear your thoughts on that. I think I should also be able to write some code and provide a PR if you're up to.

**Comment by tfausak** [comment]
You can do this with [the `render` function](https://hackage.haskell.org/package/burrito-1.1.0.2/docs/Burrito.html#v:render). For example:

``` hs
>>> import Burrito
>>> Just template = parse "http://user.example/~{name}"
>>> template
Template {tokens = ...}
>>> render template
"http://user.example/~{name}"
```

It was convenient for me to have the derived `Show` instance when I was developing this library, but now it probably makes more sense to use `render` to implement `show`. That would be similar to how `UUID` and `UTCTime` work.

**Comment by L7R7** [comment]
Oh, I totally missed `render`! That's exactly what I was looking for, thanks for the pointer.

I can imagine that it's great to have a way to see the exact structure of a given `Template` for debugging purposes. For me personally, it is good enough to use `render`, but it's probably a good idea to align the behavior of `show` to what other libraries are doing. Would you be fine with using `render` to implement `show` (which would mean that there's no direct way to print out the structure like with the derived `Show` instance)? I could provide a PR then

**Comment by tfausak** [comment]
I would definitely accept a PR that implements `show` in terms of `render`. Unfortunately I think the module structure could make that a little challenging. The `Burrito.Internal.Type.Template` module is very minimal and `Burrito.Internal.Render` depends on it. Implementing `show` in terms of `render` would require something to break that loop. 

In terms of debugging, `Template` is just a wrapper around a list of tokens. Getting the derived `show` instance would be as easy as `show (tokens template)`.

**Comment by L7R7** [comment]
I'd like to try it. Curious to see whether I can find a solution

**Comment by tfausak** [comment]
Fixed by #13.
